### PR TITLE
Add pass to print `pr.source`s to a SARIF file

### PR DESCRIPTION
**Pull request by frabert**

### CODE REVIEW

**Comment by frabert** [comment]
I had to change the name from `detect-parsers` to `vast-detect-parsers` in order for it to be recognized correctly by `lit`

File: tools/vast-detect-parsers/ParserSourceDetector.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2021-present, Trail of Bits, Inc.
```
Comment (xlauko) [inline tools/vast-detect-parsers/ParserSourceDetector.cpp:1]:
```suggestion
// Copyright (c) 2024, Trail of Bits, Inc.
```

File: tools/vast-detect-parsers/SarifPasses.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2022-present, Trail of Bits, Inc.
```
Comment (xlauko) [inline tools/vast-detect-parsers/SarifPasses.hpp:1]:
```suggestion
// Copyright (c) 2024, Trail of Bits, Inc.
```

File: tools/vast-detect-parsers/main.cpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021-present, Trail of Bits, Inc.
```
Comment (xlauko) [inline tools/vast-detect-parsers/main.cpp:1]:
```suggestion
// Copyright (c) 2024, Trail of Bits, Inc.
```

**Review by xlauko — APPROVED** [review]

**Pull request merged**